src/analysis/fepout_analysis.py

```diff
@@ -0,0 +1,611 @@
+"""
+A simple tool to automatically get results from NAMD simulation output. Author: Xufan Gao; Acknowledgement: Kevin C. Chan
+
+Test: at `Lei2Guanqiao`: python fepout_analysis.py -f D3N-N3D.csv -r D3N-N3D -c time-conv -dt 1.0
+"""
+__version__ = 0.0
+expected_states = ['bound', 'free']
+expected_keys = ['forward', 'backward']
+expected_components = ['elec', 'vdw']
+
+import warnings
+warnings.simplefilter(action='ignore', category=Warning)
+import os, shutil, sys, argparse, logging, collections
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from matplotlib.font_manager import FontProperties as FP
+import alchemlyb
+import alchemlyb_parsing_namd as apn
+from alchemlyb.estimators import MBAR, BAR, TI
+methods = {'BAR': BAR, 'MBAR': MBAR, 'TI': TI}
+from alchemlyb.postprocessors.units import to_kcalmol, R_kJmol, kJ2kcal, get_unit_converter
+k_b = R_kJmol*kJ2kcal
+# from alchemlyb.visualisation import plot_convergence
+# from alchemlyb.convergence import forward_backward_convergence
+
+
+## user IO
+class CustomMetavarFormatter(argparse.RawTextHelpFormatter):
+    """
+    Reference: https://devpress.csdn.net/python/62fe2a1dc67703293080479b.html
+
+    If the optional takes a value, format is: ``-s ARGS, --long ARGS``; Now changed to ``-s, --long ARGS``
+    """
+
+    def _format_action_invocation(self, action):
+        if not action.option_strings:
+            metavar, = self._metavar_formatter(action, action.dest)(1)
+            return metavar
+        else:
+            parts = []
+            if action.nargs == 0:
+                parts.extend(action.option_strings)
+            else:
+                default = action.dest.upper()
+                args_string = self._format_args(action, default)
+                for option_string in action.option_strings:
+                    # parts.append('%s %s' % (option_string, args_string))
+                    parts.append('%s'%option_string)
+                parts[-1] += ' %s'%args_string
+            return ', '.join(parts)
+
+
+def parse_arguments(version):
+    usage = "python fepout_analysis.py [-h] -f xxx.csv [options]"
+    des = "Automated BAR analysis for NAMD fepout files, considering both forward and backward runs.\n\n"
+    des += "An example input csv file should look like (empty lines and lines starting with # are commented out):\n"
+    des += "forward,bound,1.csv\nforward,bound,2.csv\n...\nbackward,free,3.csv\nbackward,free,4.csv\n"
```

Comment:
the format of csv example is inconsistent with what you defined later. "bound,forward,1.csv" instead.